graylog2-web-interface/src/logic/content-packs/ContentPack.js

```diff
@@ -0,0 +1,249 @@
+import { Map } from 'immutable';
+import { concat, remove } from 'lodash';
+import uuid from 'uuid/v4';
+
+export default class ContentPack {
+  constructor(v, id, rev, name, summary, description, vendor, url, requires,
+    parameters, entities) {
+    this._value = {
+      v,
+      id,
+      rev,
+      name,
+      summary,
+      description,
+      vendor,
+      url,
+      requires,
+      parameters,
+      entities,
+    };
+  }
+
+  get v() {
+    return this._value.v;
+  }
+
+  get id() {
+    return this._value.id;
+  }
+
+  get rev() {
+    return this._value.rev;
+  }
+
+  get name() {
+    return this._value.name;
+  }
+
+  get summary() {
+    return this._value.summary;
+  }
+
+  get description() {
+    return this._value.description;
+  }
+
+  get vendor() {
+    return this._value.vendor;
+  }
+
+  get url() {
+    return this._value.url;
+  }
+
+  get requires() {
+    return this._value.requires;
+  }
+
+  get parameters() {
+    return this._value.parameters;
+  }
+
+  get entities() {
+    return this._value.entities;
+  }
+
+  toBuilder() {
+    const {
+      v,
+      id,
+      rev,
+      name,
+      summary,
+      description,
+      vendor,
+      url,
+      requires,
+      parameters,
+      entities,
+    } = this._value;
+    // eslint-disable-next-line no-use-before-define
+    return new Builder(Map({
+      v,
+      id,
+      rev,
+      name,
+      summary,
+      description,
+      vendor,
+      url,
+      requires,
+      parameters,
+      entities,
+    }));
+  }
+
+  toObject() {
+    return this._value;
```

Comment:
I think we should never expose the internal state of the object directly, it makes the whole object mutable. We should either clone the state before returning it, or (way better in this case) implement a `toJSON` function that returns a new object containing the desired serialization format for JSON (As AFAIK `toObject` is only used in conjunction with `JSON.stringify`).